File: MinCostClimbingStairs.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int minCostClimbingStairs(std::vector<int>& cost) {
        const int n = cost.size();
        std::vector<int> dp(n + 1, 0);
        int i = 2;
        while (i <= n){
          dp[i] = std::min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]);
          i++;
        }
        return dp[n];
    }
};

int main(){
  Solution A;
  std::vector<int> task{1, 100, 1, 1, 1, 100, 1, 1, 100, 1};
  std::cout << A.minCostClimbingStairs(task) << '\n';
  return 0;
}

File: MinCostClimbingStairs.py
class Solution:
    def minCostClimbingStairs(self, cost: list[int]) -> int:
        n = len(cost)
        dp = [0] * (n + 1)
        i = 2
        while i <= n:
            dp[i] = min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2])
            i += 1
        return dp[n]

A = Solution()
print(A.minCostClimbingStairs(cost = [1,100,1,1,1,100,1,1,100,1]))